src/main/java/core/basesyntax/ArrayList.java

```diff
@@ -1,48 +1,155 @@
 package core.basesyntax;
 
+import java.util.NoSuchElementException;
+
 public class ArrayList<T> implements List<T> {
+    private static final int DEFAULT_CAPACITY = 10;
+    private static final int ELEMENT_NOT_FOUND_INDEX = -1;
+    private static final int EMPTY_LIST_SIZE = 0;
+    private static final int CHANGE_BY_ONE = 1;
+    private static final int ARRAY_BEGINNING_INDEX = 0;
+    private static final int DEFAULT_ARRAY_GROW_COEFFICIENT = 2;
+    private Object[] elements;
+    private int size;
+
+    public ArrayList() {
+        elements = new Object[DEFAULT_CAPACITY];
```

Comment:
Better to make array parametrized to minimize casts in methods code
```suggestion
        elements = (T[]) new Object[DEFAULT_CAPACITY];
```